test/functional/tm/ltm/test_auth.py

```diff
@@ -0,0 +1,345 @@
+            'kerberos_delegation':
+                'tm:ltm:auth:kerberos-delegation:kerberos-delegationstate',
+            'ldap': 'tm:ltm:auth:ldap:ldapstate',
+            'ocsp_responder': 'tm:ltm:auth:ocsp-responder:ocsp-responderstate',
+            'profile': 'tm:ltm:auth:profile:profilestate',
+            'radius': 'tm:ltm:auth:radius:radiusstate',
+            'radius_server': 'tm:ltm:auth:radius-server:radius-serverstate',
+            'ssl_cc_ldap': 'tm:ltm:auth:ssl-cc-ldap:ssl-cc-ldapstate',
+            'ssl_crldp': 'tm:ltm:auth:ssl-crldp:ssl-crldpstate',
+            'ssl_ocsp': 'tm:ltm:auth:ssl-ocsp:ssl-ocspstate',
+            'tacacs': 'tm:ltm:auth:tacacs:tacacsstate'
+            }
+
+    def urielementname(self):
+        if self.lowered[-2:] == '_s':
+            endind = 2
+        else:
+            endind = 1
+        return self.lowered[:-endind]
+
+    def delete_resource(self, resource):
+        try:
+            foo = resource.load(name=self.test_name, partition=self.partition)
+        except HTTPError as err:
+            if err.response.status_code != 404:
+                raise
+            return
+        foo.delete()
+
+    def setup_test(self, request, mgmt_root, **kwargs):
+        def teardown():
+            self.delete_resource(resource)
+
+        resourcecollection = \
+            getattr(getattr(getattr(mgmt_root.tm, 'ltm'), 'auth'),
+                    self.lowered)
+        resource = getattr(resourcecollection, self.urielementname())
+        self.delete_resource(resource)
+        created = resource.create(name=self.test_name,
+                                  partition=self.partition,
+                                  **kwargs)
+        request.addfinalizer(teardown)
+        return created, resourcecollection
+
+    def test_MCURDL(self, request, mgmt_root, **kwargs):
+        # Testing create
+        authres, authcollection = self.setup_test(request, mgmt_root, **kwargs)
+        assert authres.name == self.test_name
+        assert authres.fullPath == '/Common/'+self.test_name
+        assert authres.generation and isinstance(authres.generation, int)
+        assert authres.kind == self.authkinds[self.urielementname()]
+
+        # Testing update
+        authres.description = TESTDESCRIPTION
+        pp(authres.raw)
+        authres.update()
+        assert hasattr(authres, 'description')
+        assert authres.description == TESTDESCRIPTION
+
+        # Testing refresh
+        authres.description = ''
+        authres.refresh()
+        assert hasattr(authres, 'description')
+        assert authres.description == TESTDESCRIPTION
+
+        # Testing modify
+        meta_data = authres.__dict__.pop('_meta_data')
+        start_dict = copy.deepcopy(authres.__dict__)
+        authres.__dict__['_meta_data'] = meta_data
+        authres.modify(description='MODIFIED')
+        desc = 'description'
+        for k, v in authres.__dict__.items():
```

Comment:
Can you change this to iteritems(authres.__dict__) for python3 and add an "from six import iteritems" at the top of the file